be/src/http/action/compaction_action.cpp

```diff
@@ -243,7 +258,14 @@ Status CompactionAction::_execute_compaction_callback(TabletSharedPtr tablet,
         RETURN_IF_ERROR(compaction.prepare_compact());
         return compaction.execute_compact();
     };
-    if (compaction_type == PARAM_COMPACTION_BASE) {
+    if (fetch_from_remote) {
+        SingleReplicaCompaction single_compaction(_engine, tablet);
+        res = do_compact(single_compaction);
```

Comment:
Should check if single_replica compaction enabled for this table.

Reply:
check in do_compact.

Reply:
This has already been checked at line 148

Reply:
and checked in exeucte_comact()

Reply:
line 148 should_fetch_from_peer() does not check single_replica compaction enabled

Reply:
![image](https://github.com/apache/doris/assets/76934516/0c2c73b4-2ae8-48c9-b164-ff8c42ddbf03)